File: lana/src/main/scala/com/financialforce/lana/commands/Command.scala
/*
 * Copyright (c) 2020 FinancialForce.com, inc. All rights reserved.
 */
package com.financialforce.lana.commands

import com.financialforce.lana.Context
import com.financialforce.lana.runtime.vscode.commands

import scala.scalajs.js

class Command(private val name: String, private val run: js.Function) {
  private val commandPrefix = "lana."

  def register(c: Context): this.type = {
    val fullName = commandPrefix + name
    val command = commands.registerCommand(fullName, run)
    c.context.subscriptions += command
    this
  }
}

object Command {
  def apply(name: String, run: js.Function): Command = new Command(name, run)
}

File: ext/src/commands/Command.ts
/*
 * Copyright (c) 2020 FinancialForce.com, inc. All rights reserved.
 */

import { commands } from "vscode";
import { Context } from "../Context";

export class Command {
  private static commandPrefix = "lana.";

  name: string;
  run: (...args: any[]) => any;

  constructor(name: string, run: (...args: any[]) => any) {
    this.name = name;
    this.run = run;
  }

  register(c: Context): Command {
    const fullName = Command.commandPrefix + this.name;
    const command = commands.registerCommand(fullName, run);
    c.context.subscriptions.push(command);
    return this;
  }
}
